common/types/hashes.go

```diff
@@ -1,110 +1,27 @@
-type Hash12 [hash12Length]byte
-
-// Hash32 represents the 32-byte blake3 hash of arbitrary data.
-type Hash32 [Hash32Length]byte
-
-// Hash20 represents the 20-byte blake3 hash of arbitrary data.
-type Hash20 [hash20Length]byte
-
-// Field returns a log field. Implements the LoggableField interface.
-func (h Hash12) Field() log.Field { return log.String("hash", hex.EncodeToString(h[:])) }
-
-// Bytes gets the byte representation of the underlying hash.
-func (h Hash20) Bytes() []byte { return h[:] }
-
-// Big converts a hash to a big integer.
-func (h Hash20) Big() *big.Int { return new(big.Int).SetBytes(h[:]) }
-
-// Hex converts a hash to a hex string.
-func (h Hash20) Hex() string { return util.Encode(h[:]) }
-
-// String implements the stringer interface and is used also by the logger when
-// doing full logging into a file.
-func (h Hash20) String() string {
-	return h.Hex()
-}
-
-// ShortString returns a the first 5 hex-encoded bytes of the hash, for logging purposes.
-func (h Hash20) ShortString() string {
-	return hex.EncodeToString(h[:5])
-}
-
-// Format implements fmt.Formatter, forcing the byte slice to be formatted as is,
-// without going through the stringer interface used for logging.
-func (h Hash20) Format(s fmt.State, c rune) {
-	_, _ = fmt.Fprintf(s, "%"+string(c), h[:])
-}
-
-// UnmarshalText parses a hash in hex syntax.
-func (h *Hash20) UnmarshalText(input []byte) error {
-	if err := util.UnmarshalFixedText("Hash", input, h[:]); err != nil {
-		return fmt.Errorf("unmarshal text: %w", err)
-	}
-	return nil
-}
-
-// UnmarshalJSON parses a hash in hex syntax.
-func (h *Hash20) UnmarshalJSON(input []byte) error {
-	if err := util.UnmarshalFixedJSON(hashT, input, h[:]); err != nil {
-		return fmt.Errorf("unmarshal JSON: %w", err)
-	}
-
-	return nil
-}
-
-// MarshalText returns the hex representation of h.
-func (h Hash20) MarshalText() ([]byte, error) {
-	return util.Bytes(h[:]).MarshalText()
-}
-
-// SetBytes sets the hash to the value of b.
-// If b is larger than len(h), b will be cropped from the left.
-func (h *Hash20) SetBytes(b []byte) {
-	if len(b) > len(h) {
-		b = b[len(b)-32:]
-	}
-
-	copy(h[32-len(b):], b)
-}
-
-// ToHash32 returns a Hash32 whose first 20 bytes are the bytes of this Hash20, it is right-padded with zeros.
-func (h Hash20) ToHash32() (h32 Hash32) {
-	copy(h32[:], h[:])
-	return
-}
-
-// Field returns a log field. Implements the LoggableField interface.
-func (h Hash20) Field() log.Field { return log.String("hash", hex.EncodeToString(h[:])) }
+const (
+	Hash20Length = primitive.Hash20Length
+	Hash32Length = primitive.Hash32Length
+)
 
-// CalcHash12 returns the 12-byte prefix of the blake3 sum of the given byte slice.
-func CalcHash12(data []byte) (h Hash12) {
-	h32 := hash.Sum(data)
-	copy(h[:], h32[:])
-	return
-}
+var (
+	BytesToHash = primitive.BytesToHash
+	CalcHash20  = primitive.CalcHash20
+	CalcHash32  = primitive.CalcHash32
+	HexToHash32 = primitive.HexToHash32
+)
```

Comment:
Re-exports are a symptomatic fix of having types are in packages where they don't belong 🙂